src/layer.rs

```diff
@@ -36,6 +36,12 @@ where
         )
     }
 
+    /// Returns the inner `tracing_suscriber`'s [`Layer`](tracing_subscriber::layer::Layer), in
+    /// case wrapping it into a [`Registry`](registry::Registry) is undesired.
+    pub fn inner(self) -> impl layer::Layer<S> {
+        self.0
+    }
```

Comment:
This should probably be moved to the last impl block with unrestriced generics, otherwise it won't be usable as soon as the layer stops using the default configuration.
I'd also prefer providing the full type rather than an impl Trait, given it's available and unlikely to change.